src/commands/signIn.ts

```diff
@@ -1,12 +1,20 @@
 import * as vscode from "vscode";
 import {getSession} from "../auth/auth";
+import {canReachGitHubAPI} from "../api/canReachGitHubAPI";
+import {getGitHubContext} from "../git/repository";
 
 export function registerSignIn(context: vscode.ExtensionContext) {
   context.subscriptions.push(
     vscode.commands.registerCommand("github-actions.sign-in", async () => {
       const session = await getSession(true);
       if (session) {
+        const canReachAPI = await canReachGitHubAPI();
+        const ghContext = await getGitHubContext();
+        const hasGitHubRepos = ghContext && ghContext.repos.length > 0;
+
         await vscode.commands.executeCommand("setContext", "github-actions.signed-in", true);
+        await vscode.commands.executeCommand("setContext", "github-actions.internet-access", canReachAPI);
+        await vscode.commands.executeCommand("setContext", "github-actions.has-repos", hasGitHubRepos);
```

Comment:
It might be worth updating the `has-repos` context on refresh too

https://github.com/github/vscode-github-actions/blob/a1b382a9f970ce4cfd979d5dacb20b8709d44e15/src/treeViews/treeViews.ts#L24-L34

Reply:
Sounds great, we just added it there too!